dm/config/source_config.go

```diff
@@ -158,20 +158,28 @@ func (c *SourceConfig) Yaml() (string, error) {
 	return string(b), nil
 }
 
-// Parse parses flag definitions from the argument list.
+// FromToml parses flag definitions from the argument list.
 // accept toml content for legacy use (mainly used by etcd).
-func (c *SourceConfig) Parse(content string) error {
+func (c *SourceConfig) FromToml(content string) error {
 	// Parse first to get config file.
 	metaData, err := toml.Decode(content, c)
-	err2 := c.check(&metaData, err)
-	if err2 != nil {
-		return err2
+	if err != nil {
+		return terror.ErrWorkerDecodeConfigFromFile.Delegate(err)
+	}
+	undecoded := metaData.Undecoded()
+	if len(undecoded) > 0 && err == nil {
```

Comment:
```suggestion
	if len(undecoded) > 0 {
```

seems always err is always nil